File: robotpy/RobotPy-WPILib/samples/KitBase/robot.py
import wpilib

lstick = wpilib.Joystick(1)
rstick = wpilib.Joystick(2)

lmotor = wpilib.CANJaguar(3)
rmotor = wpilib.CANJaguar(5)
drive = wpilib.RobotDrive(lmotor, rmotor)

def CheckRestart():
    if lstick.GetRawButton(10):
        raise RuntimeError("Restart")

class MyRobot(wpilib.SimpleRobot):
    def Disabled(self):
        while self.IsDisabled():
            CheckRestart()
            wpilib.Wait(0.01)

    def Autonomous(self):
        self.GetWatchdog().SetEnabled(False)
        while self.IsAutonomous() and self.IsEnabled():
            CheckRestart()
            wpilib.Wait(0.01)

    def OperatorControl(self):
        dog = self.GetWatchdog()
        dog.SetEnabled(True)
        dog.SetExpiration(0.5)

        while self.IsOperatorControl() and self.IsEnabled():
            dog.Feed()
            CheckRestart()

            drive.TankDrive(lstick, rstick)

            wpilib.Wait(0.04)

def run():
    robot = MyRobot()
    robot.StartCompetition()

File: SummerDrivetrain/src/stuy/Robot.java
/*----------------------------------------------------------------------------*/
/* Copyright (c) FIRST 2008. All Rights Reserved.                             */
/* Open Source Software - may be modified and shared by FRC teams. The code   */
/* must be accompanied by the FIRST BSD license file in the root directory of */
/* the project.                                                               */
/*----------------------------------------------------------------------------*/
package stuy;

import edu.wpi.first.wpilibj.SimpleRobot;
import edu.wpi.first.wpilibj.CANJaguar;
import edu.wpi.first.wpilibj.RobotDrive;
import edu.wpi.first.wpilibj.Joystick;

/**
 * The VM is configured to automatically run this class, and to call the
 * functions corresponding to each mode, as described in the SimpleRobot
 * documentation. If you change the name of this class or the package after
 * creating this project, you must also update the manifest file in the resource
 * directory.
 */
public class Robot extends SimpleRobot implements Constants {

    CANJaguar frontLeftMotor, rearLeftMotor, frontRightMotor, rearRightMotor;
    RobotDrive drivetrain;
    Joystick gamepad;

    public Robot() {
        try {
            frontLeftMotor = new CANJaguar(DRIVE_CAN_DEVICE_FRONT_LEFT);
            rearLeftMotor = new CANJaguar(DRIVE_CAN_DEVICE_REAR_LEFT);
            frontRightMotor = new CANJaguar(DRIVE_CAN_DEVICE_FRONT_RIGHT);
            rearRightMotor = new CANJaguar(DRIVE_CAN_DEVICE_REAR_RIGHT);
        }
        catch (Exception e) {
        }

        drivetrain = new RobotDrive(frontLeftMotor, rearLeftMotor, frontRightMotor, rearRightMotor);

        // TODO: Check for inverted motors.
        drivetrain.setInvertedMotor(RobotDrive.MotorType.kFrontLeft, false);
        drivetrain.setInvertedMotor(RobotDrive.MotorType.kRearLeft, false);
        drivetrain.setInvertedMotor(RobotDrive.MotorType.kFrontRight, false);
        drivetrain.setInvertedMotor(RobotDrive.MotorType.kRearRight, false);

        gamepad = new Joystick(PORT_GAMEPAD);
    }

    /**
     * This function is called once each time the robot enters autonomous mode.
     */
    public void autonomous() {
    }

    /**
     * This function is called once each time the robot enters operator control.
     */
    public void operatorControl() {
        getWatchdog().setEnabled(false);
        while (isEnabled() && isOperatorControl()) {
            drivetrain.tankDrive(gamepad, 2, gamepad, 4);
        }
    }
}
